### PR TITLE
Config with pure python

### PR DESCRIPTION
**Pull request by icemac**
* Lint the code
* Add support for Python 3.9
* Remove unused non-BBB imports

### CODE REVIEW

**Comment by icemac** [comment]
Coverage might have decreased because of now testing branch coverage only on Python 3.

**Review by mgedmin — COMMENTED** [review]
I'd say LGTM, except I'm really uneasy about the potential backwards compatibility issues with IFormField.prefix:

- the default used to be '' (no prefix), and I'm not sure the default in the interface actually matters somewhere, but it makes me uneasy to change it nonetheless
- the constraint doesn't allow periods, but I have used `zope.formlib.form.Fields(...,  prefix='foo.')` in my code, and now I'm afraid that code will break.

(zope.formlib itself uses periods in form field names e.g. as the self.name + `.used` suffix; I'm not sure why they're disallowed by the constraint.)

File: src/zope/formlib/interfaces.py [hunk 1]
```diff
@@ -310,19 +324,20 @@ class ITextBrowserWidget(ISimpleInputWidget):
...
+
 def reConstraint(pat, explanation):
     pat = re.compile(pat)
 
     def constraint(value):
-        if prefix_re.match(value):
+        if pat.match(value):
             return True
         raise Invalid(value, explanation)
+    return constraint
```
[the remark this answers is not in the document]
Comment (icemac) [reply src/zope/formlib/interfaces.py]: Restored the old behavior in https://github.com/zopefoundation/zope.formlib/pull/34/commits/cebfb9be32505f98546ba1159ce7e8e8c3aabf7c.

File: src/zope/formlib/interfaces.py [hunk 2]
```diff
@@ -859,31 +876,31 @@ class IFormField(Interface):
         disambiguate fields with the same name (e.g. from different
         schema) within a collection of form fields.
         """,
-        default="",
-        )
+        default="form_field",
```
[the remark this answers is not in the document]
Comment (icemac) [reply src/zope/formlib/interfaces.py]: Restored the old behavior in https://github.com/zopefoundation/zope.formlib/pull/34/commits/cebfb9be32505f98546ba1159ce7e8e8c3aabf7c.

File: src/zope/formlib/tests/test_form.py [hunk 3]
```diff
@@ -26,32 +26,29 @@ def _makeContext(self, **kw):
         class _Context(object):
             pass
         context = _Context()
-        for k, v in kw.items():
-            setattr(context, k, v)
```
[the remark this answers is not in the document]
Comment (icemac) [reply src/zope/formlib/tests/test_form.py:30]: Fixed in https://github.com/zopefoundation/zope.formlib/pull/34/commits/d0b7b0666cd0f9c1d2b17aa7e308e4a7911e2e8a

**Comment by icemac** [comment]
Moved fixing `reConstraint` to https://github.com/zopefoundation/zope.formlib/pull/35.

Thank you for reviewing this PR. 😃

**Pull request merged**